docs/source/cpp/flight.rst

```diff
@@ -239,7 +239,19 @@ Memory management
 -----------------
 
 Flight tries to reuse allocations made by gRPC to avoid redundant
-data copies. However, this means that those allocations may not
+data copies. However, experience shows that such data is frequently
+misaligned. Some use cases might require data to have data type-specific
+alignment (for example, for the data buffer of an Int32 array to be aligned
+on a 4-byte boundary), which can be enforced
+by setting :member:`arrow::ipc::IpcReadOptions::ensure_alignment`
+to :member:`arrow::ipc::Alignment::kDataTypeSpecificAlignment`.
+This uses the :member:`arrow::ipc::IpcReadOptions::memory_pool`
+to a allocate memory with aligned addresses, but only for mis-alligned data.
+However, this creates data copies of your data recieved via Flight.
+
+.. note:: Ensuring memory alignment is not supported for non-CPU Flight data.
```

Comment:
Do we need this note? Does Flight even support non-CPU data? That information isn't preserved accross IPC.

Reply:
Right, that might imply non-CPU Flight data exists. Will remove.